frontend/src/components/QueryBuilderV2/QueryV2/MetricsSelect/MetricsSelect.tsx

```diff
@@ -16,19 +17,28 @@ export const MetricsSelect = memo(function MetricsSelect({
 	version: string;
 	signalSource: 'meter' | '';
 }): JSX.Element {
+	const [attributeKeys, setAttributeKeys] = useState<BaseAutocompleteData[]>([]);
+
 	const { handleChangeAggregatorAttribute } = useQueryOperations({
 		index,
 		query,
 		entityVersion: version,
 	});
 
+	const handleAggregatorAttributeChange = useCallback(
+		(value: BaseAutocompleteData, isEditMode?: boolean) => {
+			handleChangeAggregatorAttribute(value, isEditMode, attributeKeys || []);
+		},
+		[handleChangeAggregatorAttribute, attributeKeys],
+	);
 	return (
 		<div className="metrics-select-container">
 			<AggregatorFilter
-				onChange={handleChangeAggregatorAttribute}
+				onChange={handleAggregatorAttributeChange}
 				query={query}
 				index={index}
 				signalSource={signalSource || ''}
+				setAttributeKeys={setAttributeKeys}
```

Comment:
Just wanted to check. There are multiple places where AggregatorFilter is used. This is the only place where this change is required right?